src/pages/index.js

```diff
@@ -1,52 +1,158 @@
-import {
-  initialCards,
-  validationOptions,
-  buttons,
-} from "../utils/constants.js";
+import { validationOptions, buttons } from "../utils/constants.js";
 import FormValidator from "../components/FormValidator.js";
 import Card from "../components/Card.js";
 import Popup from "../components/Popup.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
 import "./index.css";
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", (cardData) => {
-    previewImagePopup.open({ link: cardData.link, name: cardData.name });
-  });
-  return card.getView();
-}
-
-// Instances
+// Popups
 
 const editProfilePopup = new PopupWithForm({
   popupSelector: "#edit-profile-modal",
   handleFormSubmit: (data) => {
-    userInfo.setUserInfo(data);
+    editProfilePopup.setLoading(true, "Saving...");
+
+    api
+      .updateUserProfile(data.userName, data.userDescription)
+      .then((updatedData) => {
+        userInfo.setUserInfo({
+          userName: updatedData.name,
+          userDescription: updatedData.about,
+        });
+        editProfilePopup.close();
+      })
+      .catch((err) => console.error("Error updating user profile:", err))
+      .finally(() => {
+        editProfilePopup.setLoading(false);
+      });
+  },
+});
+
+const updateAvatarPopup = new PopupWithForm({
+  popupSelector: "#update-avatar-modal",
+  handleFormSubmit: (data) => {
+    const avatarLink = data.link;
+    updateAvatarPopup.setLoading(true, "Saving...");
+
+    api
+      .updateUserAvatar(avatarLink)
+      .then((updatedData) => {
+        const profileAvatar = document.querySelector(".profile__avatar");
+        profileAvatar.src = updatedData.avatar;
```

Comment:
`        profileAvatar.src = updatedData.avatar;
`
Only methods of `UserInfo` should set/get the profile data (including the **avatar**)